### DISCUSSION TITLE
change USERNAME_FIELD with extending

### DISCUSSION THREAD

**Discussion by saeedfarhang** [saleor/saleor#10373] [Deployment]
hi there. i was wondering if i could change USERNAME_FIELD for account.User model without changing the base code and just by extending saleor. i want to add phone number for username. is there any way or best practices?

thanks for the great work on saleor by the way.

**Comment by karolkielecki** [comment]
Hi @saeedfarhang, first of all -- thanks for your kind words! What exactly would you like to achieve?

`USERNAME_FIELD` determines which field will be the user's unique identifier. In Saleor, this is an email by default. You can fork and modify the code to your needs by adding a phone number, but if you do that, staying up to date with future Saleor releases will be difficult. 

Perhaps, metadata can also be helpful in such a case: https://docs.saleor.io/docs/3.x/developer/metadata

Cheers,
